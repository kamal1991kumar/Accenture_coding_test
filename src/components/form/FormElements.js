import React from "react";
import PropTypes from "prop-types";
import InputField from "./inputField";
import TextArea from "./textArea";

export const FromElements = (props) => {
  const { type } = props;
  let View = InputField;
  if (type === "textarea") {
    View = TextArea;
  }
  return <View {...props} />;
};

FromElements.propTypes = {
  type: PropTypes.string.isRequired,
};
