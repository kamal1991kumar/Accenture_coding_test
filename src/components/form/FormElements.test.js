import React from "react";
import { render } from "@testing-library/react";
import { FromElements } from "./FormElements";

const props = {
  hasError: false,
  message: "input not be empty",
  value: "text",
  handleChange: jest.fn(),
  type: "input",
};

describe("FormElemets Component", () => {
  test("Should render inputField component", () => {
    const { getByTestId } = render(<FromElements {...props} />);
    expect(getByTestId("input")).toMatchSnapshot();
  });

  test("Should render textArea component", () => {
    const { getByTestId } = render(<FromElements {...props} type="textarea" />);
    expect(getByTestId("textArea")).toMatchSnapshot();
  });
});
