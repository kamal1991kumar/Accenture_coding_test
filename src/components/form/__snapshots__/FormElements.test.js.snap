// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormElemets Component Should render inputField component 1`] = `
<input
  class="inputFiled__input inputFiled__input-"
  data-testid="input"
  type="text"
  value="text"
/>
`;

exports[`FormElemets Component Should render textArea component 1`] = `
<textarea
  class="textArea__input textArea__input-"
  data-testid="textArea"
>
  text
</textarea>
`;
